examples/nlp/language_modeling/megatron_retro_eval.py

```diff
@@ -105,35 +211,18 @@ def main(cfg) -> None:
         "add_BOS": cfg.inference.add_BOS,
         "all_probs": cfg.inference.all_probs,
         "compute_logprob": cfg.inference.compute_logprob,
+        "end_strings": cfg.inference.end_strings,
     }
 
-    # check whether the DDP is initialized
-    if not parallel_state.is_initialized():
-
-        def dummy():
-            return
-
-        if model.trainer.strategy.launcher is not None:
-            model.trainer.strategy.launcher.launch(dummy, trainer=model.trainer)
-        model.trainer.strategy.setup_environment()
-
+    # Second method of running text generation, call trainer.predict [recommended]
+    prompt = [cfg.prompt]
+    neighbors = [cfg.neighbors]
+    ds = RequestDataSet(prompt, neighbors)
+    bs = 1
+    request_dl = DataLoader(dataset=ds, batch_size=bs)
     config = OmegaConf.to_container(cfg.inference)
-    retrieval_service = OmegaConf.to_container(cfg.retrieval_service)
-    model.set_inference_config(config, retrieval_service)
-
-    if not cfg.use_predict_method:
-        # First method of running text generation, call model.generate method
-        response = model.generate(
-            inputs=OmegaConf.to_container(cfg.prompts),
-            length_params=length_params,
-            sampling_params=sampling_params,
-            strategy=model.inference_strategy,
-        )
-    else:
-        # Second method of running text generation, call trainer.predict
-        ds = RequestDataSet(OmegaConf.to_container(cfg.prompts))
-        request_dl = DataLoader(dataset=ds, batch_size=cfg.inference_batch_size)
-        response = trainer.predict(model, request_dl)
+    model.set_inference_config(config)
+    response = trainer.predict(model, request_dl)
```

Comment:
Why use trainer.predict ? Why not keep it consistant with all the other inference code and use model.generate()

Reply:
I actually learned from other LM files, and find that trainer.predict is used and recommended. For example:
 - [megatron_gpt_eval.py](https://github.com/NVIDIA/NeMo/blob/38657465c81aa19c30d63d85603da650ee98c949/examples/nlp/language_modeling/megatron_gpt_eval.py#L328)
 - [megatron_t5_eval.py](https://github.com/NVIDIA/NeMo/blob/38657465c81aa19c30d63d85603da650ee98c949/examples/nlp/language_modeling/megatron_t5_eval.py#L136)

Reply:
I see, thanks . I didnt see that before.